sshuttle/methods/tproxy.py

```diff
@@ -194,11 +204,11 @@ def _ipt_proto_ports(proto, fport, lport):
                  '-m', 'udp', '-p', 'udp')
 
         for _, ip in [i for i in nslist if i[0] == family]:
-            _ipt('-A', mark_chain, '-j', 'MARK', '--set-mark', '1',
+            _ipt('-A', mark_chain, '-j', 'MARK', '--set-mark', tmark,
                  '--dest', '%s/32' % ip,
                  '-m', 'udp', '-p', 'udp', '--dport', '53')
             _ipt('-A', tproxy_chain, '-j', 'TPROXY',
-                 '--tproxy-mark', '0x1/0x1',
+                 '--tproxy-mark', '0x'+tmark+'/0x'+tmark,
```

Comment:
I think there's a bug here: eg if tmark was '10', then --set-mark would set the mark to 10 for locally generated packets, but tproxy-mark would set the mark to 0x10 (i.e. 16) for packets arriving from the local network.

Also: the part after the slash is the mask, and doesn't have to be equal to the mark. tmark should probably default to `1/1`, and be used directly without any `0x` prefix. That way the flag can specify the value and mask.

Reply:
Thank you for that comment @normanr , you're right. That should be improved here just as you mention. Maybe better to create a new PR since this is already closed.